includes/wikia/tests/TitleBatchTest.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+class TitleBatchTest extends WikiaBaseTest {
+
+	public function setUp() {
+		parent::setUp();
+	}
+
+	// This test cover constructor, getArticlesIds and check exact behaviour of getWikiaProperties
+	public function testGetWikiaPropertiesShouldOnlyReturnPropertiesForTitlesThatContains() {
+
+		$expectedProperty = 17;
+		$expectedId1 = 1;
+		$expectedId2 = 124;
+		$expectedId3 = 1243;
+
+		$expectedPropValue1 = 'a';
+		$expectedPropValue2 = 'b';
+		$expectedPropValue3 = null;
+
+		$expectedIds = [ $expectedId1, $expectedId2, $expectedId3 ];
+
+		$titles = [ ];
+		foreach ( $expectedIds as $titleId ) {
+			$titles[] = $this->getTestTitle( $titleId );
+		}
+
+		$mockDb = $this->getMockBuilder( '\DatabaseMysql' )
+			->disableOriginalConstructor()
+			->setMethods( [ 'select' ] )
+			->getMock();
+
+		$fields = [ 'p' => 'page', 'pp' => 'page_wikia_props' ];
+		$table = [ 'p.page_id, pp.propname, pp.props' ];
+		$cond = [ 'p.page_id = pp.page_id',
+			'pp.propname' => $expectedProperty,
+			'p.page_id' => $expectedIds ];
+
+		$result = [
+			(object)[ 'page_id' => $expectedId1, 'propname' => $expectedProperty, 'props' => serialize( $expectedPropValue1 ) ],
+			(object)[ 'page_id' => $expectedId2, 'propname' => $expectedProperty, 'props' => serialize( $expectedPropValue2 ) ],
+		];
+
+		$mockDb->expects( $this->once() )
+			->method( 'select' )
+			->with( $fields, $table, $cond, 'TitleBatch::getWikiaProperties' )
+			->will( $this->returnValue( $result ) );
+
+		$mockGetDB = $this->getGlobalFunctionMock( 'wfGetDB' );
```

Comment:
Can be simplified if the check for DB_SLAVE is not relevant by replacing the entire block by a single line:
```
$this->mockGlobalFunction('wfGetDB',$mockDb);
```